core/src/main/java/org/elasticsearch/common/ParseField.java

```diff
@@ -30,36 +30,33 @@
 
     private static final DeprecationLogger DEPRECATION_LOGGER = new DeprecationLogger(Loggers.getLogger(ParseField.class));
 
-    private final String camelCaseName;
     private final String underscoreName;
     private final String[] deprecatedNames;
     private String allReplacedWith = null;
 
     public ParseField(String value, String... deprecatedNames) {
-        camelCaseName = Strings.toCamelCase(value);
         underscoreName = Strings.toUnderscoreCase(value);
-        if (deprecatedNames == null || deprecatedNames.length == 0) {
-            this.deprecatedNames = Strings.EMPTY_ARRAY;
-        } else {
-            final HashSet<String> set = new HashSet<>();
-            for (String depName : deprecatedNames) {
-                set.add(Strings.toCamelCase(depName));
-                set.add(Strings.toUnderscoreCase(depName));
-            }
-            this.deprecatedNames = set.toArray(new String[set.size()]);
+        final HashSet<String> set = new HashSet<>();
+        String camelCaseName = Strings.toCamelCase(value);
+        if (camelCaseName.equals(value) == false) {
+            set.add(camelCaseName);
+        }
+        for (String depName : deprecatedNames) {
+            set.add(Strings.toCamelCase(depName));
+            set.add(Strings.toUnderscoreCase(depName));
         }
```

Comment:
It's not exactly user error right? It's plugin author error? I'm fine with that in 2.x, but I'd rather keep the code as close to original as possible for pushing into 2.3.2?